.github/workflows/discover-lean-pr-testing.yml

```
@@ -122,21 +122,84 @@ jobs:
           echo "branches_exist=true" >> "$GITHUB_ENV"
         fi
 
+    - name: Execute merge script for each branch
+      id: execute-merges
+      if: env.branches_exist == 'true'
+      run: |
+        BRANCHES="${{ steps.find-branches.outputs.branches }}"
+
+        # Initialize arrays to track results
+        SUCCESSFUL_MERGES=""
+        FAILED_MERGES=""
+
+        # Ensure the merge script is executable
+        chmod +x scripts/merge-lean-testing-pr.sh
+
+        # Process each branch
+        for BRANCH in $BRANCHES; do
+          # Extract PR number from branch name
+          PR_NUMBER=$(echo "$BRANCH" | grep -oP '\d+$')
+          GITHUB_DIFF="https://github.com/leanprover-community/mathlib4/compare/nightly-testing...lean-pr-testing-$PR_NUMBER"
+
+          echo "Attempting to merge branch: $BRANCH (PR #$PR_NUMBER)"
+
+          # Save the current state to restore in case of failure
+          git reset --hard HEAD
+          git checkout nightly-testing
```

I don't understand the comment.
```suggestion
          # Reset to a clean state
          git reset --hard HEAD
          git checkout nightly-testing
```